### ISSUE TITLE
Fix S3168: Rule should not be too strict on first parameter name

### ISSUE THREAD

**Issue by fahadabdulaziz** [SonarSource/sonar-csharp#1161]
### Description

Hi, 
Thank you for the great job you done still doing.. 

I have little issue with analyzer S3168, it reports false positive about events handlers has async to change it to returns Task instead of void. it's impossible to change it as you know.

### Repro steps

1. Create WinFroms project
2. Add a button to the form
3. Double click on the button.
4. Add a new method that return Task.
5. Call that method returns task from the button's event handler. and you will get the problem.

### Expected behavior

It should not report the event handler, while the analyzer is really really useful with other methods.

### Actual behavior

It reports on the event handlers that uses async.

### Known workarounds

No workaround known.

### Related information

* SonarC# Version: 6.8.1.4648
* Visual Studio Version: 15.5.5

Thank you

**Comment by Evangelink** [comment]
Hi @fahadabdulaziz,

Glad to hear you are enjoying our work :)

Regarding the issue you are talking about, I cannot reproduce the error you are experimenting. Could you create a small reproducer? Are you using the NuGet or SonarLint for VS?

As you can see `button1_Click` doesn't raise an issue while `Foo` does.
![image](https://user-images.githubusercontent.com/11340282/36089416-1b73cef4-0fdc-11e8-9f79-65abc11451b1.png)

**Comment by fahadabdulaziz** [comment]
I got the error in this code:

```csharp
public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private async void button1_Click(object __sender, EventArgs __e) // got the problem here
        {
            await DoTask().ConfigureAwait(true);
        }

        public async Task DoTask()
        {
        }
    }
```

The problem is we name our methods parameters to has (__) as prefix and that what cause the analyzer to not detect that the method is event handler

**Comment by Evangelink** [comment]
The second parameter name is not a problem (we only check that it is or it derives from `EventArgs`) but for the first one we check that the type is `object` and that it is called `sender`. I'll discuss the problem with the team and I'll get back to you right after.

**Comment by fahadabdulaziz** [comment]
I see you fixed this issue in the code, any idea when we could get the update.

Thank you

**Comment by Evangelink** [comment]
Hi @fahadabdulaziz,

We will release 7.0 on Monday.

Cheers,
Amaury

**Comment by fahadabdulaziz** [comment]
Hi,
Thank you,
